pyked/converters.py

```
@@ -0,0 +1,741 @@
+
+    def __str__(self):
+        return repr('Error: Keyword not defined: {}'.format(self.keywords[0]))
+
+
+def get_file_metadata(root):
+    """Read and parse ReSpecTh XML file metadata (file author, version, etc.)
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with file metadata
+    """
+    properties = {}
+
+    properties['file-author'] = {'name': ''}
+    try:
+        properties['file-author']['name'] = root.find('fileAuthor').text
+    except AttributeError:
+        raise AttributeError('Error: no fileAuthor given')
+
+    if properties['file-author']['name'] == '':
+        raise AttributeError('Error: no fileAuthor given')
+
+    # Default version is 0
+    version = '0'
+    elem = root.find('fileVersion')
+    if elem is None:
+        print('Warning: no fileVersion given')
+    else:
+        try:
+            version = elem.find('major').text + '.' + elem.find('minor').text
+        except AttributeError:
+            print('Error: missing fileVersion major/minor')
+            raise
+    properties['file-version'] = int(float(version))
+
+    # Default ChemKED version
+    properties['chemked-version'] = __version__
+
+    return properties
+
+
+def get_reference(root):
+    """Read reference info from root of ReSpecTh XML file.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with reference information
+    """
+    reference = {}
+    elem = root.find('bibliographyLink')
+    if elem is None:
+        raise AttributeError('Error: missing bibliographyLink')
+
+    # Try to get reference info via DOI
+    try:
+        reference['doi'] = elem.attrib['doi']
+        ref = None
+        try:
+            ref = habanero.Crossref().works(ids=reference['doi'])['message']
+        except (HTTPError, habanero.RequestError):
+            print('DOI not found')
+            raise KeyError
```

> this `KeyError` is raised so that the outer except catches it

Right, I'm suggesting to use one of our custom errors, in case someone has intercepted `KeyError`s somewhere upstream. Although, given your second comment, its maybe better to just raise the appropriate error here, rather than relying on the outer `except` to catch and re-raise? There's probably also a way to simplify this logic that doesn't rely on nested try-except statements (I was pretty confused the first time I read through this).